extensions/amp-install-serviceworker/0.1/amp-install-serviceworker.js

```diff
@@ -325,15 +330,52 @@ function sendAmpScriptToSwOnFirstVisit(registration, win) {
           .filter(item => item.initiatorType === 'script' &&
             item.name.indexOf(urls.cdn) !== -1)
           .map(script => script.name);
-      // using https://github.com/redux-utilities/flux-standard-action
+      // using convention from https://github.com/redux-utilities/flux-standard-action.
       controllerSw.postMessage(JSON.stringify(dict({
-        'type': 'FIRST_VISIT_CACHING',
+        'type': 'AMP__FIRST-VISIT-CACHING',
         'payload': ampScriptsUsed,
       })));
     });
   }
 }
 
+/**
+ * Whenever a new service worker is activated, controlled page will send
+ * the used AMP scripts and the self's URL to service worker to be cached.
+ * @param {ServiceWorkerRegistration} registration
+ * @param {!Window} win
+ */
+function prefetchOutgoingLinks(registration, win) {
+  const {document} = win;
+  const links = [].map.call(document.querySelectorAll('a[data-rel=prefetch]'),
+      link => link.getAttribute('href'));
```

Comment:
This won't be the normalized href. You want `.href` not `.getAttribute('href')`

Reply:
gotcha